pkg/apiclient/apiclient.go

```diff
@@ -27,6 +27,7 @@ type Client interface {
 
 type Opts struct {
 	ArgoServerOpts ArgoServerOpts
+	Context *context.Context
```

Comment:
should not be pointer type

Reply:
if you add a `GetContext` func, then if `Context` is not set, you could just return `context.Background()`